packages/@next-fetch/react-query/src/form.ts

```diff
@@ -32,15 +32,31 @@ export function useForm<Data, Error, Input, Context>(
  * This enables progressive enhancement, as the form can be submitted
  * without having to re-render the app using JavaScript code.
  */
-export function Form<Data, Error, Input, Context>({
-  mutation,
-  mutationConfig,
-  ...props
-}: React.HTMLProps<HTMLFormElement> &
+
+// 4. https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref/58473012#58473012
+declare module "react" {
+  function forwardRef<T, P = {}>(
+    render: (props: P, ref: React.ForwardedRef<T>) => React.ReactElement | null
+  ): (props: P & React.RefAttributes<T>) => React.ReactElement | null;
+}
```

Comment:
I worry about this changing customer code because AFAIK TypeScript declarations are global. Maybe we should "unsafely" type-cast?

```ts
// typed-forward-ref.ts

import { forwardRef as forwardRef_ } from "react";
const forwardRef = forwardRef_ as unknown as TYPEDEF_HERE;
export default forwardRef;
```

wdyt? 👀